source/v0/SocketAddr.cpp

```diff
@@ -35,34 +35,36 @@ bool SocketAddr::is_v4() {
 
 #define OCTET_SIZE 3
 #define SEPARATOR_SIZE 1
-#define TERMINATOR_SIZE 1
-#define IPv4_STRLEN (4 * (OCTET_SIZE) + 3 * SEPARATOR_SIZE + TERMINATOR_SIZE)
+#define IPv4_STRLEN (4 * (OCTET_SIZE) + 3 * SEPARATOR_SIZE)
 #define IPv6_QUAD_SIZE 4
 #define IPv64_PREFIX ("::ffff:")
-#define IPv64_PREFIX_STRLEN (sizeof(IPv64_PREFIX))
+#define IPv64_PREFIX_STRLEN (sizeof(IPv64_PREFIX) - 1)
 #define IPv64_STRLEN (IPv64_PREFIX_STRLEN + IPv4_STRLEN)
 
 
 // Returns 0 on success
 int SocketAddr::fmtIPv4(char *buf, size_t size)
 {
-    if (size < IPv4_STRLEN) {
-        return -1;
-    }
+    if (buf == NULL) return -1;
+    if (size <= IPv4_STRLEN) return -2;
```

Comment:
If there's going to be more than one error code, then it should be an enum. Using -1, -2, -3 is very brittle and hard to debug.